### PR TITLE
[ncp] add support for SRP client in NCP and spinel

### PR DESCRIPTION
**Pull request by abtink**
This commit adds new spinel definitions for SRP client support
(mirroring the public OT APIs of SRP client) and implements get, set,
add, and remove handlers in `Ncp` for the new spinel properties.

-------

This is currently pushed on top of commits from PR #6038. 
_**Please just check/review the last commit.**_

### CODE REVIEW

File: src/lib/spinel/spinel.h [hunk 1]
```diff
@@ -718,6 +718,32 @@ enum
...
+    SPINEL_SRP_CLIENT_ITEM_STATE_REFRESHING = 3, // Item is being refreshed.
+    SPINEL_SRP_CLIENT_ITEM_STATE_TO_REMOVE  = 4, // Item to be removed.
+    SPINEL_SRP_CLIENT_ITEM_STATE_REMOVING   = 5, // Item is being removed.
+    SPINEL_SRP_CLIENT_ITEM_STATE_REGISTERED = 6, // Item is registered with server.
+    SPINEL_SRP_CLIENT_ITEM_STATE_REMOVED    = 7, // Item is removed.
+} spinel_srp_client_item_state_t;
+
+typedef enum
+{
+    SPINEL_SRP_CLIENT_ERROR_NONE             = 0, // No error
```
Comment (jwhui) [inline src/lib/spinel/spinel.h:735]: End lines with period to be consistent with above?
```suggestion
    SPINEL_SRP_CLIENT_ERROR_NONE             = 0, // No error.
```

File: src/lib/spinel/spinel.h [hunk 2]
```diff
@@ -3637,6 +3664,143 @@ enum
...
+     *
+     * Writing to this property allows user to start or stop the SRP client operation with a given SRP server.
+     *
+     * Written value format is:
+     *
+     *   `b` : TRUE to start the client, FALSE to stop the client.
+     *
+     * When used to start the SRP client, the following fields should also be included:
+     *
+     *   `6` : SRP server IPv6 address (only when SRP client to be started).
```
Comment (jwhui) [inline src/lib/spinel/spinel.h:3679]: "only when SRP client to be started" seems to be redundant with above.

File: src/lib/spinel/spinel.h [hunk 3]
```diff
@@ -3637,6 +3664,143 @@ enum
...
+     *   `U` : SRP server port number (only when SRP client to be started).
+     *   `b` : Boolean to indicate whether or not to emit SRP client events (using `SPINEL_PROP_SRP_CLIENT_EVENT`).
```
Comment (jwhui) [inline src/lib/spinel/spinel.h:3681]: Is this also "only when SRP client to be started"?

File: src/lib/spinel/spinel.h [hunk 4]
```diff
@@ -3637,6 +3664,143 @@ enum
...
+     *   `t(A(6))` : Structure containing array of host IPv6 addresses.
+     *
+     * The active or removed services data is an array of services `A(t(UUSSSd))` with each service format:
+     *
+     *   `U` : The service name labels (e.g., "_chip._udp", not the full domain name.
+     *   `U` : The service instance name label (not the full name).
+     *   `S` : The service port number.
+     *   `S` : The service priority.
+     *   `S` : The service weight.
+     *   `d` : The encoded TXT-DATA
```
Comment (jwhui) [inline src/lib/spinel/spinel.h:3799]:
```suggestion
     *   `d` : The encoded TXT-DATA.
```

File: src/lib/spinel/spinel.h (hunk 3 above)
Comment (abtink) [reply src/lib/spinel/spinel.h:3681]: Yes. We only receive events from SRP client when it is started.

**Review by jwhui — APPROVED** [review]

**Pull request merged**